src/app/demos/affix/affix-demo.component.html

```diff
@@ -0,0 +1,152 @@
+<sky-page>
+  <div
+    class="canvas"
+  >
+    <div
+      class="parent-scrollable"
+      id="screenshot-affix"
+      [ngClass]="{
+        'enable-scroll': enableScrollableParent
+      }"
+      #parentScrollable
+    >
+      <div
+        class="parent-scrollable-canvas"
+      >
+        <div
+          class="target"
+          style="width:50px;height:50px;"
+          #target
+        >
+          <div
+            class="target target-gap"
+            style="width:40px;height:40px;"
+          >
+            <div
+              class="target"
+              style="width:30px;height:30px;"
+            >
+              <div
+                class="target target-gap"
+                style="width:20px;height:20px;"
+              >
+                <div
+                  class="target"
+                  style="width:10px;height:10px;"
+                ></div>
+              </div>
+            </div>
+          </div>
+        </div>
+      </div>
+    </div>
+
+    <div
+      class="subject"
+      [affixEnableAutoFit]="enableAutoFit"
+      [affixHorizontalAlignment]="model.horizontalAlignment"
+      [affixIsSticky]="isSticky"
+      [affixPlacement]="model.placement"
+      [affixVerticalAlignment]="model.verticalAlignment"
+      [ngClass]="{ 'subject-hidden': !isVisible }"
```

Comment:
Are we expecting consumers to handle visibility? Why couldn't the service handle this?

Reply:
Yes, the consumer should handle visibility themselves. The consumer may wish to keep showing the subject element even if the target is hidden (or do something else, like make the subject element full screen). I just provided an event to alert the consumer when the target element is hidden from view.

Reply:
Ok, I can totally get behind that. That does bring up another question though - Are the terms "subject" and "target" the best choices here? They don't inherently communicate which is which. Is there some pairing of words out there that would better explain the hierarchy? Parent/child? Host/child? etc...?